### PR TITLE
MAINT: Sort dependencies alphabetically

### PR DESCRIPTION
**Pull request by oesteban**
Splitting up #2959 in digestible bits.

This would be great to merge quickly if possible.

## Summary

Fixes # .

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

**Comment by effigies** [comment]
This is fine by me, but I think the idea was to give a rough order of perceived importance of the dependency. Is the change just to make it easier to search and see if a dependency is or isn't in the list?

File: nipype/info.py [hunk 1]
```diff
@@ -135,47 +135,47 @@ def get_nipype_gitversion():
...
-    'networkx>=%s,<=%s ; python_version < "3.0"' % (NETWORKX_MIN_VERSION, NETWORKX_MAX_VERSION_27),
+    'click>=%s' % CLICK_MIN_VERSION,
+    'configparser; python_version <= "3.4"',
+    'funcsigs',
+    'future>=%s' % FUTURE_MIN_VERSION,
+    'futures; python_version == "2.7"',
     'networkx>=%s ; python_version >= "3.0"' % NETWORKX_MIN_VERSION,
-    'numpy>=%s,!=%s ; python_version == "2.7"' % (NUMPY_MIN_VERSION, NUMPY_BAD_VERSION_27),
+    'networkx>=%s,<=%s ; python_version < "3.0"' % (NETWORKX_MIN_VERSION, NETWORKX_MAX_VERSION_27),
+    'neurdflib',
```
Comment (satra) [inline nipype/info.py]: this one has to come at the end. we are using pip install order hack to install a same named library but with additions that remain to be merged into a release.

File: requirements.txt [hunk 2]
```diff
@@ -1,16 +1,16 @@
-numpy>=1.9.0
-scipy>=0.14
+click>=6.6.0
+configparser
+funcsigs
+future>=0.16.0
 networkx>=1.9
-traits>=4.6
-python-dateutil>=2.2
+neurdflib
```
Comment (satra) [inline requirements.txt]: same here. this has to come over prov.

**Comment by oesteban** [comment]
> Is the change just to make it easier to search and see if a dependency is or isn't in the list?

Yes, that is the idea.

@satra, I'd say to remove those dependencies in a separate PR, just in case tests start failing...

@kaczmarj any fast way to instruct neurodocker to run all pip installs with ``--no-cache-dir``?

**Comment by satra** [comment]
> I'd say to remove those dependencies in a separate PR, just in case tests start failing

unfortunately i don't think we have a test for this. which i'll try to add. however this is not about removing, but the order of listing the dependencies. so `neurdflib` **has to come after** `prov`. 

right now the tests are failing for other reasons.

**Comment by effigies** [comment]
This might be fixed by using a newer neurodocker.

**Comment by kaczmarj** [comment]
@oesteban - neurodocker should include some `conda clean` call after `conda install`. at least version 0.5.0 does. is the cache present in the container?

**Comment by oesteban** [comment]
@kaczmarj 0.5.0 is failing too

**Comment by kaczmarj** [comment]
@oesteban - i will try to look into this tomorrow

**Comment by effigies** [comment]
The issue is caused by https://github.com/conda/conda-package-handling/pull/34. We need to set the working directory to something owned by the `neuro` user.

File: requirements.txt (hunk 2 above)
Comment (effigies) [reply requirements.txt]:
```suggestion
```

File: requirements.txt [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 nibabel>=2.1.0
-future>=0.16.0
-simplejson>=3.8.0
+numpy>=1.9.0
+packaging
 prov>=1.5.2
```
Comment (effigies) [inline requirements.txt]:
```suggestion
prov>=1.5.2
neurdflib
```

File: rtd_requirements.txt [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 networkx>=1.9
-traits>=4.6
-python-dateutil>=2.2
 nibabel>=2.1.0
-future>=0.16.0
-simplejson>=3.8.0
+numpy>=1.9.0
+numpydoc
+packaging
 prov==1.5.0
```
Comment (effigies) [inline rtd_requirements.txt]:
```suggestion
prov>=1.5.2
neurdflib
```

**Comment by oesteban** [comment]
@effigies, I added your suggestions within the merge upstream/master commit. I hope that's fine.

File: rtd_requirements.txt [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
+future>=0.16.0
 matplotlib
-packaging
+mock
+networkx>=1.9
+nibabel>=2.1.0
+numpy>=1.9.0
 numpydoc
+packaging
+prov>=1.5.2
```
Comment (effigies) [inline rtd_requirements.txt]:
```suggestion
prov>=1.5.2
neurdflib
```

**Comment by satra** [comment]
@effigies - looks good to me. the one thing about rtd is that it was limited to the dependencies required for doc building, but it's fine to have everything. we should definitely check, once this is merged that rtd does build.

**Comment by effigies** [comment]
Docs built fine. https://readthedocs.org/projects/nipype/builds/9387007/

Somewhat unclear why they aren't built with every commit to master. I had to manually trigger the build.

**Comment by satra** [comment]
i clicked on a few buttons (specifically enabled github integration - which i thought was already enabled). let's see if that makes a difference.

**Pull request merged**